hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/StreamSync.java

```diff
@@ -453,6 +454,7 @@ HoodieTableMetaClient initializeEmptyTable(HoodieTableMetaClient.TableBuilder ta
             Boolean.parseBoolean(HIVE_STYLE_PARTITIONING_ENABLE.defaultValue())))
         .setUrlEncodePartitioning(props.getBoolean(URL_ENCODE_PARTITIONING.key(),
             Boolean.parseBoolean(URL_ENCODE_PARTITIONING.defaultValue())))
+        .setPluggableTableFormatClass(props.getProperty(HoodieTableConfig.SUPPLEMENTARY_TABLE_FORMAT_CLASS_NAME.key(), HudiPluggableTableFormat.class.getName()))
```

Comment:
There are other spark writers apart from deltastreamer/spark-sql, planning to support them as well in the future?

Reply:
For tests, I think we should implement a `TestTableFormat`? 

and +1 on making all write paths and read paths respect this.